src/components/SiteCarousel.js

```diff
@@ -0,0 +1,151 @@
+// 3rd party libraries - core
+import React, {Component} from 'react';
+import {View, StyleSheet, ActivityIndicator, Dimensions, ImageBackground, TouchableOpacity} from 'react-native';
+import {Icon, Card, Text} from 'react-native-elements';
+import Swiper from 'react-native-swiper';
+import _ from 'lodash';
+
+import SitePreview from './SitePreview';
+
+// 3rd party libraries - additional
+
+// styles and language
+import {campsite} from '../locale.en';
+import {attemptToAddFavorite, attemptToRemoveFavorite, getSiteDetail} from "../actions";
+import {connect} from "react-redux";
+
+const {campsite_form: {accessibility_options}} = campsite;
+
+const SCREEN_WIDTH = Dimensions.get('window').width;
+
+
+class SiteCarousel extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {}
+    }
+
+    componentWillMount() {
+        const {displaySites, selectedSite} = this.props;
+
+        if(displaySites.length === 0){
+            return;
+        }
+
+        const currentSiteIndex = _.findIndex(displaySites, site => site.id === selectedSite.id);
+
+        this.setState({
+            initialIndex: currentSiteIndex > -1 ? currentSiteIndex : 0
+        });
+    }
+
+    componentWillReceiveProps(nextProps) {
+        const {selectedSite} = this.props;
+
+        if(nextProps.displaySites.length === 0){
+            return;
+        }
+
+        if(selectedSite.id !== nextProps.selectedSite.id){
+            let currentSiteIndex = _.findIndex(nextProps.displaySites, site => site.id === nextProps.selectedSite.id);
+
+            this.setState({
+                initialIndex: currentSiteIndex > -1 ? currentSiteIndex : 0
+            });
```

Comment:
Whenever the carousel loads or whenever it receives props, check to see if we have `displaySites`, if we do, check if we have a `selectedSite` and set our starting index in the carousel to that position.

Reply:
Note this also works to reset the index when we select a new site on the map once the carousel is already open.